.github/workflows/doc-publish.yml

```diff
@@ -0,0 +1,23 @@
+# documentation publish
+---
+name: doc-publish
+on:
+  push:
+    branches:
+      #- master
+      #- main
```

Comment:
Does this mean it doesn't run at all if they are commented out?

Reply:
No, it works for any branch, for testing purpose this has been done. Removed.